File: SilentCaster-Electron/src/app/services/response.service.ts
import { Injectable } from '@angular/core';
import { QuickResponse, QuickResponseImpl } from '../models/quick-response.model';

@Injectable({
  providedIn: 'root'
})
export class ResponseService {
  private responses: QuickResponse[] = [];
  private readonly RESPONSES_KEY = 'quick_responses';

  constructor() {
    this.loadResponses();
  }

  getAllResponses(): QuickResponse[] {
    return [...this.responses];
  }

  getResponsesForMessage(message: string): string[] {
    const responses: string[] = [];

    // Ищем совпадения по триггерам
    for (const response of this.responses) {
      if (!response.isEnabled) continue;
      
      if (response.trigger === '*' || 
          message.toLowerCase().includes(response.trigger.toLowerCase())) {
        responses.push(...response.responses);
      }
    }

    // Удаляем дубликаты
    return [...new Set(responses)];
  }

  getPersonalResponsesForMessage(message: string): string[] {
    const responses: string[] = [];

    // Ищем только персональные ответы (isQuickResponse == false)
    for (const response of this.responses) {
      if (!response.isEnabled || response.isQuickResponse) continue;
      
      if (response.trigger === '*' || 
          message.toLowerCase().includes(response.trigger.toLowerCase())) {
        responses.push(...response.responses);
      }
    }

    return [...new Set(responses)];
  }

  addResponse(response: QuickResponse): void {
    this.responses.push(response);
    this.saveResponses();
  }

  removeResponse(index: number): void {
    if (index >= 0 && index < this.responses.length) {
      this.responses.splice(index, 1);
      this.saveResponses();
    }
  }

  updateResponse(index: number, response: QuickResponse): void {
    if (index >= 0 && index < this.responses.length) {
      this.responses[index] = response;
      this.saveResponses();
    }
  }

  clearResponses(): void {
    this.responses = [];
    this.saveResponses();
  }

  getQuickResponses(): QuickResponse[] {
    return this.responses.filter(r => r.isQuickResponse);
  }

  getPersonalResponses(): QuickResponse[] {
    return this.responses.filter(r => !r.isQuickResponse);
  }

  updateAllResponses(newResponses: QuickResponse[]): void {
    this.responses = [...newResponses];
    this.saveResponses();
  }

  private loadResponses(): void {
    try {
      if (window.electronAPI && (window.electronAPI as any).loadResponses) {
        const saved = (window.electronAPI as any).loadResponses();
        if (saved && Array.isArray(saved)) {
          this.responses = saved;
          return;
        }
      }

      // Fallback to localStorage
      const saved = localStorage.getItem(this.RESPONSES_KEY);
      if (saved) {
        this.responses = JSON.parse(saved);
      } else {
        // Создаем примеры
        this.responses = [
          { ...new QuickResponseImpl(), trigger: 'привет', responses: ['Привет всем!', 'Приветики!'], isQuickResponse: true },
          { ...new QuickResponseImpl(), trigger: 'спасибо', responses: ['Пожалуйста!', 'Рад помочь!'], isQuickResponse: true },
          { ...new QuickResponseImpl(), trigger: 'follow', responses: ['Спасибо за фоллоу, {username}!'], isQuickResponse: false },
          { ...new QuickResponseImpl(), trigger: 'привет, {username}', responses: ['Привет, {username}!'], isQuickResponse: false }
        ];
        this.saveResponses();
      }
    } catch (error) {
      console.error('Error loading responses:', error);
      this.responses = [];
    }
  }

  private saveResponses(): void {
    try {
      if (window.electronAPI && (window.electronAPI as any).saveResponses) {
        (window.electronAPI as any).saveResponses(this.responses);
        return;
      }

      // Fallback to localStorage
      localStorage.setItem(this.RESPONSES_KEY, JSON.stringify(this.responses));
    } catch (error) {
      console.error('Error saving responses:', error);
    }
  }
}

File: SilentCaster-Legacy/Services/ResponseService.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Newtonsoft.Json;
using SilentCaster.Models;

namespace SilentCaster.Services
{
    public class ResponseService
    {
        private const string ResponsesFile = "Responses.json";
        private List<QuickResponse> _responses;

        public ResponseService()
        {
            _responses = new List<QuickResponse>();
            LoadResponses();
        }

        public List<QuickResponse> GetAllResponses()
        {
            return _responses.ToList();
        }

        public List<string> GetResponsesForMessage(string message)
        {
            var responses = new List<string>();

            // 查找匹配的触发器
            foreach (var response in _responses)
            {
                if (response.Trigger == "*" || 
                    message.ToLower().Contains(response.Trigger.ToLower()))
                {
                    responses.AddRange(response.Responses);
                }
            }

            return responses.Distinct().ToList();
        }

        public List<string> GetPersonalResponsesForMessage(string message)
        {
            var responses = new List<string>();

            // Ищем только персональные ответы (IsQuickResponse == false)
            foreach (var response in _responses.Where(r => !r.IsQuickResponse))
            {
                if (response.Trigger == "*" || 
                    message.ToLower().Contains(response.Trigger.ToLower()))
                {
                    responses.AddRange(response.Responses);
                }
            }

            return responses.Distinct().ToList();
        }

        public void AddResponse(QuickResponse response)
        {
            _responses.Add(response);
            SaveResponses();
        }

        public void RemoveResponse(int index)
        {
            if (index >= 0 && index < _responses.Count)
            {
                _responses.RemoveAt(index);
                SaveResponses();
            }
        }

        public void ClearResponses()
        {
            _responses.Clear();
            SaveResponses();
        }

        public List<QuickResponse> GetQuickResponses()
        {
            return _responses.Where(r => r.IsQuickResponse).ToList();
        }

        public List<QuickResponse> GetPersonalResponses()
        {
            return _responses.Where(r => !r.IsQuickResponse).ToList();
        }

        public void UpdateAllResponses(List<QuickResponse> newResponses)
        {
            _responses.Clear();
            _responses.AddRange(newResponses);
            SaveResponses();
        }

        private void LoadResponses()
        {
            try
            {
                if (File.Exists(ResponsesFile))
                {
                    var json = File.ReadAllText(ResponsesFile);
                    _responses = JsonConvert.DeserializeObject<List<QuickResponse>>(json) ?? new List<QuickResponse>();
                }
                else
                {
                    // Создаем примеры быстрых и персональных ответов
                    _responses = new List<QuickResponse>
                    {
                        new QuickResponse { Trigger = "привет", Responses = new List<string>{"Привет всем!", "Приветики!"}, IsQuickResponse = true },
                        new QuickResponse { Trigger = "спасибо", Responses = new List<string>{"Пожалуйста!", "Рад помочь!"}, IsQuickResponse = true },
                        new QuickResponse { Trigger = "follow", Responses = new List<string>{"Спасибо за фоллоу, {username}!"}, IsQuickResponse = false },
                        new QuickResponse { Trigger = "привет, {username}", Responses = new List<string>{"Привет, {username}!"}, IsQuickResponse = false }
                    };
                    SaveResponses();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine($"Ошибка загрузки ответов: {ex.Message}");
                _responses = new List<QuickResponse>();
            }
        }

        public void SaveResponses()
        {
            try
            {
                var json = JsonConvert.SerializeObject(_responses, Formatting.Indented);
                File.WriteAllText(ResponsesFile, json);
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine($"Ошибка сохранения ответов: {ex.Message}");
            }
        }
    }
} 